modules/cloudsql-postgres/docs/auth.md

```diff
@@ -0,0 +1,68 @@
+# Authentication Model for Cloud SQL PostgreSQL Module
+
+This document explains **how workloads authenticate to Cloud SQL instances provisioned by the `cloudsql‑postgres` Terraform module**, covering both GKE and non‑Kubernetes consumers. The design relies on **Cloud SQL IAM database authentication**.
+
+## Summary
+
+* **IAM database authentication** is enabled on every instance via the `cloudsql.iam_authentication` flag ([Cloud SQL IAM auth guide](https://cloud.google.com/sql/docs/postgres/iam-authentication)).
+* **Workload Identity** binds a Kubernetes Service Account (KSA) to a Google Service Account (GSA) that has `roles/cloudsql.client`; the GSA's short‑lived IAM token becomes the database credential ([GKE Workload Identity overview](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity)).
+* Workloads can connect through the **Cloud SQL Auth Proxy** sidecar or **direct Private IP with an IAM token**; both approaches are fully supported ([Auth Proxy v2 documentation](https://cloud.google.com/sql/docs/postgres/connect-auth-proxy) • [Private‑IP connection guide](https://cloud.google.com/sql/docs/postgres/connect-instance-private-ip)).
+* The module **never creates password users**; teams who need one must add a separate `google_sql_user` resource ([Managing PostgreSQL users](https://cloud.google.com/sql/docs/postgres/create-manage-users)).
+
+## 1. GKE connection options
+
+### 1.1. Cloud SQL Auth Proxy sidecar *(recommended)*
+
+1. **Bind KSA → GSA** with Workload Identity Federation ([binding guide](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#binding_ksa_to_gsa)).
+2. Module grants the bound **GSA** `roles/cloudsql.client` ([Cloud SQL IAM roles reference](https://cloud.google.com/sql/docs/postgres/iam-roles-permissions)).
+3. Pod runs the Auth Proxy v2 sidecar:
+
+   ```yaml
+   - name: cloud-sql-proxy
+     image: gcr.io/cloud-sql-connectors/cloud-sql-proxy:2
+     args: ["--private-ip", "--auto-iam-authn", "$(INSTANCE)"]
+   ```
+
+   `--auto-iam-authn` instructs the proxy to issue a fresh IAM token for each connection ([Auth Proxy v2 documentation](https://cloud.google.com/sql/docs/postgres/connect-auth-proxy)).
+4. Application connects to `localhost:5432`; no credentials are stored in the container.
+
+### 1.2. Direct Private‑IP (no proxy)
+
+1. Generate a token with the Google Cloud CLI:
+
+   ```bash
+   TOKEN=$(gcloud auth print-access-token)
+   ```
+2. Connect using `psql`:
+
+   ```bash
+   psql "host=${PRIVATE_IP} sslmode=require \
+         user=iam:${GSA_EMAIL} password=${TOKEN} dbname=myapp"
+   ```
+
+   The `iam:` prefix tells Cloud SQL to validate the OAuth 2.0 token ([IAM login syntax](https://cloud.google.com/sql/docs/postgres/iam-logins)).
```

Comment:
considering removing this, as we avoid using tokens whenever possible

if you do decide to keep it, consider hiding it under a collapsed section
https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections

and add that you need the role `roles/cloudsql.instanceUser`

Reply:
removed it